spec/README.md

```diff
@@ -1,133 +1,61 @@
-
-#### Layers:
-
-The content layer always consists of the WASM module binary. 
-
-The config layer consists of a JSON-formatted string, which contains metadata for the target runtime. The runtime and ABI (Application Binary Interface) versions of an image can be deduced by parsing the config layer. 
-
-The config layer may also contain additional data, depending on the target runtime. For example, the config for a WASM Envoy Filter contains root_ids available on the filter. 
-
-For the sake of simplicity, the specification only supports a single module per image.
-
-#### Running OCI Images with Envoy:
-
-Envoy supports loading and running WASM modules via a file on local disk or an “Http datasource”.
+| Wasm Module                        | The distributable, loadable, and executable unit of code in WebAssembly. 
+| Wasm Image Specification           | The specification for storing Wasm modules as container images.
+| Wasm Runtime                       | The execution environment into which a Wasm Module may be loaded. This refers to the application itself which loads and executes a wasm module. Examples include web browsers, the Open Policy Agent, the Envoy Proxy, or any other application which supports extension via Wasm modules. 
+| Runtime Configuation              | Configuration or metadata specific to the runtime which consumes a module. 
 
-Envoy WASM Filters can be stored according to the spec and run with Istio and Gloo, with the help of a local cache which pulls images from remote registries.  
+## Specifications
 
-Control planes then configure the Envoy instances to load the filter via the local cache, using the required root_id parameter supplied in the image config if it is available.
+Here we have several specifications for how to bundle Wasm modules as container images. 
 
+There are two variants of the specification:
+- [spec.md](spec.md)
+- [spec-compat.md](spec-compat.md)
 
-### Format:
+Developers and Wasm module consumers can leverage both of these specifications. 
 
-The WASM OCI Artifact Spec consists of two layers bundled together:
-- A layer specifying configuration for the target runtime
-- A layer containing the compiled WASM module itself
+For clarity, we call the variant in [spec.md](spec.md) *oci*, and the one in [spec-compat.md](spec-compat.md) *compat*.
 
-Each layer is associated with its own Media Type, which is stored in the OCI Descriptor for that layer:
+## Difference between variants
 
-| Media Type | Type | Description |
-|------------|------|-------------|
-| application/vnd.module.wasm.config.v1+json | JSON Object | Configuration for the Target WASM runtime.
-| application/vnd.module.wasm.content.layer.v1+wasm | binary data (byte array) | The compiled module data |
+The key difference between these two variants is that, the *oci* variant makes use of the custom media types on [OCI Artifact] for image layers while the *compat* variant leverages the standard media types.
```

Comment:
Maybe we should add a sentence like "our goal is to make the OCI variant the default format for shipping Wasm modules in container images, we acknowledge however that there are toolchains and registries deployed and in use that do not support our custom media types yet. To accomodate those exisiting toolchains, there is the semantically equivalent *compat* variant, which provides the same feature set, but is compatible with existing tooling because it 'looks' very much like a normal container image. Implementations of this spec should support both variants."

Reply:
done! https://github.com/solo-io/wasm/pull/265/commits/d06e5880695665da98d90f770a9c741ba94ba2fa